src/Ocelot.Provider.Polly/PollyQoSProvider.cs

```diff
@@ -63,14 +66,20 @@ private PollyPolicyWrapper<HttpResponseMessage> PollyPolicyWrapperFactory(Downst
                 .Or<TimeoutException>()
                 .CircuitBreakerAsync(route.QosOptions.ExceptionsAllowedBeforeBreaking,
                     durationOfBreak: TimeSpan.FromMilliseconds(route.QosOptions.DurationOfBreak),
-                    onBreak: (ex, breakDelay) => _logger.LogError(info + $"Breaking the circuit for {breakDelay.TotalMilliseconds} ms!", ex.Exception),
+                    onBreak: (ex, breakDelay) =>
+                        _logger.LogError(info + $"Breaking the circuit for {breakDelay.TotalMilliseconds} ms!",
+                            ex.Exception),
                     onReset: () => _logger.LogDebug(info + "Call OK! Closed the circuit again."),
                     onHalfOpen: () => _logger.LogDebug(info + "Half-open; Next call is a trial."));
         }
 
+        // No default set for polly timeout at the minute.
+        // Since a user could potentially set timeout value = 0, we need to handle this case.
+        // TODO throw an exception if the user sets timeout value = 0 or at least return a warning
+        // TODO the design in DelegatingHandlerHandlerFactory should be reviewed
         var timeoutPolicy = Policy
             .TimeoutAsync<HttpResponseMessage>(
-                TimeSpan.FromMilliseconds(route.QosOptions.TimeoutValue), 
+                TimeSpan.FromMilliseconds(route.QosOptions.TimeoutValue),
                 TimeoutStrategy.Pessimistic);
```

Comment:
Ah-ha-h!.. Do we have `Optimistic` strategy? 😄

Reply:
this param does not exist in new Ocelot Pipelin

Reply:
😹